### PR TITLE
Make local session coordination transparent

### PR DESCRIPTION
**Pull request by Kahtaf**
## Summary

- Replace user-facing writer/follower behavior with authenticated local coordination across GUI/TUI sessions.
- Add session-scoped action IDs, durable accepted/pending dedupe, lock migration/recovery hardening, and neutral reconnect/sync UX.
- Proxy supported chat, tool, and ask_user actions to the active local coordinator, with safeguards for TUI-owned sessions and admission/retry edge cases.
- Update OpenSpec/docs/changelog plus verification artifacts for the transparent coordinator flow.

## Verified screenshots

GUI home

GUI catalog

GUI sidebar

## Verification

- `npm test` -> latest full pass: 217 files, 2285 tests passed.
- `npm run test:gui:browser` -> 23/23 passed in Chromium, including coordinated multi-client GUI smoke coverage.
- `npm run gui:web:build` -> passed; Vite reported the existing >500 kB chunk warning.
- `openspec validate transparent-local-session-coordinator --strict` -> passed.
- `graphify update .` -> passed.
- TUI harness: `npx tsx tests/harness/cli.ts run --prompt "Answer briefly: what is 2+2?" --ipc validation-output/transparent-local-session-coordinator/tui-harness-ipc --timeout 180000` -> finalText `4`, turns `1`, interactions `0`, duration `2710ms`.
- In-app Browser snapshots checked for absence of user-facing `writer`, `follower`, `read-only`, and `takeover` wording.
- `npm run review:pr` -> clean, no accepted/actionable findings; warning: review bundle was truncated, so coverage is partial. Parallel review gate ran `npx tsc --noEmit && npx vitest run` successfully.

## Notes

- TUI-owned routed sessions support forwarded chat prompts; non-chat GUI actions remain hidden/unavailable for TUI-owned sessions until matching TUI coordinator endpoints exist.
- Verification artifacts live under `docs/verification/transparent-local-session-coordinator/`.

### CODE REVIEW

File: gui/web/src/App.jsx [hunk 1]
```diff
@@ -97,16 +98,26 @@ export function AppShell() {
...
   });
   const liveEvents = liveEventsBySession[sessionView.activeSessionId] || [];
   const liveBaseEventCount = liveBaseEventCountBySession[sessionView.activeSessionId] || 0;
-  const visibleAskUserPrompts = gui.askUserPrompts.filter(
-    (prompt) => !prompt.sessionId || prompt.sessionId === sessionView.activeSessionId,
-  );
+  const nonChatActionsUnavailable =
+    gui.coordination?.sessionId === sessionView.activeSessionId &&
+    gui.coordination?.ownerKind === "tui" &&
+    gui.role !== "writer";
```
[the remark this answers is not in the document]
Comment (Kahtaf) [reply gui/web/src/App.jsx]: Verified against the current branch: `nonChatActionsUnavailable` now keys only on the active session's coordination (`gui.coordination?.sessionId === sessionView.activeSessionId && gui.coordination?.ownerKind === "tui"`) with no process-role term, so a writer GUI viewing a TUI-owned session hides ask-user/tool UI correctly. Two related hardening fixes landed on top: blocked tool invokes now reject instead of returning a bare `false` (so market-state mutations no longer report false success), and state snapshots now refresh coordination mid-run so the gate also engages when ownership changes after boot.

File: gui/server/ws-hub.ts [hunk 2]
```diff
@@ -329,6 +342,25 @@ export function createWsHub({
...
+function coordinationStateFor(sessionId: string, role: string) {
+  const status =
+    role === "writer"
+      ? "ready"
+      : role === "connecting"
+        ? "connecting"
+        : role === "disconnected"
+          ? "reconnecting"
+          : "syncing";
+  return { sessionId, status };
```
[the remark this answers is not in the document]
Comment (Kahtaf) [reply gui/server/ws-hub.ts]: Verified against the current branch: `coordinationStateForSession` in gui/server/ws-hub.ts includes `ownerKind` (from the session lock's `processKind`) in both the boot payload and `buildBootstrapPayload`, while `publicWriterLock` still strips only the coordinator secret — so a GUI starting on a TUI-owned session gates non-chat actions from the first boot frame. Additionally, `buildStateSnapshot` now re-derives coordination on every `state.snapshot` broadcast and the browser applies it for the session it is tracking, covering mid-run ownership changes that boot-time state could not.
